### PR TITLE
[RFC] legacy test script: string wrapping improvement

### PR DESCRIPTION
**Pull request by glindstedt**
I added an extra check to see if a string contains `]` and then wrap it using `[=[...]=]` instead. Only using square brackets would sometimes break, for example with `[[a = b['hello']]]`. This can be useful when migrating the larger legacy tests that contain python code.

I also changed insert() and expect() to use `[=[...]=]`.

### CODE REVIEW

File: scripts/legacy2luatest.pl [hunk 1]
```diff
@@ -133,8 +133,15 @@ sub read_in_file {
         # If line contains single quotes or backslashes, use double
         # square brackets to wrap string.
         if (/'/ || /\\/) {
-          $startstr = '[[';
-          $endstr = ']]';
+            # If line contains ending square bracket, use square brackets
```
Comment (justinmk) [inline scripts/legacy2luatest.pl]: Assuming I understand this comment, I suggest rewording it:

```
# If the line contains a closing bracket without a corresponding open bracket,
# wrap it with [=[...]=].
```

**Comment by justinmk** [comment]
c @fwalch

File: scripts/legacy2luatest.pl (hunk 1 above)
Comment (glindste) [inline scripts/legacy2luatest.pl]: It doesn't matter if an opening bracket exists or not, the closing bracket can still cause a premature end of string. But I will reword the comment, thanks.

**Comment by justinmk** [comment]
is there a reason not to *always* use `[=[`?

**Comment by glindste** [comment]
Not really, the only reasons are aesthetical and possibly readability. If you think it's a good idea I can make it default.

**Comment by justinmk** [comment]
merged, thanks
